dhis2/dhis2-core
# fix: job progress - clean complete after cancel [DHIS2-13125]

### Summary
When a job is cancelled eventually a `CancellationException` is thrown. That skips execution of the job but not necessarily completes the recorded process record by calling one of the `failedProcess` methods on `JobProgress`.
As a consequence the `Notifier` never recognizes that the final message was received (run is not "complete") for the run of the job and the UI ḱeeps showing the `...` indicator.

A similar issue was that the outcome of the `JobConfiguration` could be registered as "success" in case no exception was thrown as a result of local error handling. The state was simply not based on the tracked information but exceptions alone. So even if the tracking records a cancelled or failed process the job configuration could save this as "successful" as no exception ended the execution.
To address this the set state is now based on the tracked records. 

Indirectly this also addresses an issue that when tracking is used wrongly (methods not called in pairs) this could create a second process record at the end. While this is a programming error this is still annoying from the users point of view and it can be avoided. The changed code kind of detects this and does not create these phantom process records.  

### Manual Testing
* open data administration app
* goto analytics
* run analytics job
* cancel it using a `POST` to `/api/scheduling/cancel/ANALYTICS_TABLE`
* check that the cancellation eventually is recognised by the notifier feed shown on the page
* similarly `/api/jobConfigurations` and look for the `ANALYTICS_TABLE` configuration and its `lastExecutedStatus`
